ravenframework/utils/TreeStructure.py

```diff
@@ -62,13 +58,20 @@ def parse(inFile,dType=None):
       #Possibly we should just try parsing the file instead of checking?
       raise InputParsingError('Unrecognized file type for:',inFile,' | Expected .xml')
   if dType.lower()=='xml':
-    #try:
-    parser = ET.XMLParser(target=CommentedTreeBuilder())
-    xmltree = ET.parse(inFile,parser=parser) #parser is defined below, under XMLCommentParser
-    tree = xmlToInputTree(xmltree)
-    #except Exception as e:
-    #  print('ERROR: Input parsing error!')
-    #  raise e
+    try:
+      parser = ET.XMLParser(target=CommentedTreeBuilder())
+      xmltree = ET.parse(inFile,parser=parser) #parser is defined below, under XMLCommentParser
+      tree = xmlToInputTree(xmltree)
+    except ET.ParseError as err:
+      fileName = inFile.name
+      inFile.close()
+      lineNo, col = err.position
+      with open(fileName, 'r') as inFile:
+        content = inFile.readlines()
+      line = content[lineNo-1].strip('\n')
+      caret = '{:=>{}}'.format('^', col)
+      err.msg = '{}\n{}\n{}\n in input file: {}'.format(err, line, caret, fileName)
+      raise err
```

Comment:
👏

Reply:
Could we place this checking in a common location, such as the xmlUtils, so that we can use this error whenever we parse an XML tree? This is an excellent expansion of the existing error.